src/Content/Text.php

```diff
@@ -0,0 +1,1714 @@
+                $ret["title"] = L10n::t('link to source');
+            }
+
+        } elseif (x($item, 'plink') && ($item['private'] != 1)) {
+            $ret = [
+                    'href' => $item['plink'],
+                    'orig' => $item['plink'],
+                    'title' => L10n::t('link to source'),
+                ];
+        } else {
+            $ret = [];
+        }
+
+        return $ret;
+    }
+
+    /**
+     * replace html amp entity with amp char
+     * @param string $s
+     * @return string
+     */
+    function unamp($s) {
+        return str_replace('&amp;', '&', $s);
+    }
+
+
+    /**
+     * return number of bytes in size (K, M, G)
+     * @param string $size_str
+     * @return number
+     */
+    function returnBytes($size_str) {
+        switch (substr ($size_str, -1)) {
+            case 'M': case 'm': return (int)$size_str * 1048576;
+            case 'K': case 'k': return (int)$size_str * 1024;
+            case 'G': case 'g': return (int)$size_str * 1073741824;
+            default: return $size_str;
+        }
+    }
+
+    /**
+     * @param string $s
+     * @param boolean $strip_padding
+     * @return string
+     */
+    function base64UrlEncode($s, $strip_padding = false) {
+
+        $s = strtr(base64_encode($s), '+/', '-_');
+
+        if ($strip_padding) {
+            $s = str_replace('=','',$s);
+        }
+
+        return $s;
+    }
+
+    /**
+     * @param string $s
+     * @return string
+     */
+    function base64UrlDecode($s) {
+
+        if (is_array($s)) {
+            self::logger('Text::base64UrlDecode: illegal input: ' . print_r(debug_backtrace(), true));
+            return $s;
+        }
+
+    /*
+    *  // Placeholder for new rev of salmon which strips base64 padding.
+    *  // PHP base64_decode handles the un-padded input without requiring this step
+    *  // Uncomment if you find you need it.
+    *
+    *	$l = strlen($s);
+    *	if (!strpos($s,'=')) {
+    *		$m = $l % 4;
+    *		if ($m == 2)
+    *			$s .= '==';
+    *		if ($m == 3)
+    *			$s .= '=';
+    *	}
+    *
+    */
+
+        return base64_decode(strtr($s,'-_','+/'));
+    }
+
+    /**
+     * return div element with class 'clear'
+     * @return string
+     * @deprecated
+     */
+    function clearDiv() {
+        return '<div class="clear"></div>';
+    }
+
+    function bbTranslateVideo($s) {
+
+        $matches = null;
+        $r = preg_match_all("/\[video\](.*?)\[\/video\]/ism",$s,$matches,PREG_SET_ORDER);
+        if ($r) {
+            foreach ($matches as $mtch) {
+                if ((stristr($mtch[1], 'youtube')) || (stristr($mtch[1], 'youtu.be'))) {
+                    $s = str_replace($mtch[0], '[youtube]' . $mtch[1] . '[/youtube]', $s);
+                } elseif (stristr($mtch[1], 'vimeo')) {
+                    $s = str_replace($mtch[0], '[vimeo]' . $mtch[1] . '[/vimeo]', $s);
+                }
+            }
+        }
+        return $s;
+    }
+
+    function HTML2BBVideo($s) {
```

Comment:
Code standards: Please add a new line before class method opening brace.

Reply:
Additionally, this method should be moved to the `Friendica\Content\text\HTML` class.